packages/expo/src/hooks/useAppleSignIn.ts

```diff
@@ -0,0 +1,159 @@
+import { useSignIn, useSignUp } from '@clerk/clerk-react';
+import type { SetActive, SignInResource, SignUpResource } from '@clerk/types';
+import * as AppleAuthentication from 'expo-apple-authentication';
+import { Platform } from 'react-native';
+
+import { errorThrower } from '../utils/errors';
+
+export type StartAppleSignInFlowParams = {
+  unsafeMetadata?: SignUpUnsafeMetadata;
+};
+
+export type StartAppleSignInFlowReturnType = {
+  createdSessionId: string | null;
+  setActive?: SetActive;
+  signIn?: SignInResource;
+  signUp?: SignUpResource;
+};
+
+/**
+ * Hook for native Apple Sign-In on iOS using expo-apple-authentication.
+ *
+ * This hook provides a simplified way to authenticate users with their Apple ID
+ * using the native iOS Sign in with Apple UI. The authentication flow automatically
+ * handles the ID token exchange with Clerk's backend and manages the transfer flow
+ * between sign-in and sign-up.
+ *
+ * @example
+ * ```tsx
+ * import { useAppleSignIn } from '@clerk/clerk-expo';
+ * import { Button } from 'react-native';
+ *
+ * function AppleSignInButton() {
+ *   const { startAppleSignInFlow } = useAppleSignIn();
+ *
+ *   const onPress = async () => {
+ *     try {
+ *       const { createdSessionId, setActive } = await startAppleSignInFlow();
+ *
+ *       if (createdSessionId && setActive) {
+ *         await setActive({ session: createdSessionId });
+ *       }
+ *     } catch (err) {
+ *       console.error('Apple Sign-In error:', err);
+ *     }
+ *   };
+ *
+ *   return <Button title="Sign in with Apple" onPress={onPress} />;
+ * }
+ * ```
+ *
+ * @requires expo-apple-authentication - Must be installed as a peer dependency
+ * @platform iOS - This hook only works on iOS. On other platforms, it will throw an error.
+ *
+ * @returns An object containing the `startAppleSignInFlow` function
+ */
+export function useAppleSignIn() {
```

Comment:
what do you think about renaming this to `useAppleAuthentication`? Would be easier to make it work for dedicated sign-ups (outside of the sign-in to sign-up transfer that it already supports).